index.html

```diff
@@ -161,123 +114,130 @@ <h2>
+          <li>Create a new <a>PerformanceMark</a> object (<var>entry</var>).</li>
+          <li>Set <var>entry</var>'s <code>name</code> attribute to <var>markName</var>.</li>
+          <li>Set <var>entry</var>'s <code>entryType</code> attribute to <code>DOMString "mark"</code>.</li>
+          <li>Set <var>entry</var>'s <code>startTime</code> attribute to the value that would be returned by the <code>Performance</code> object's <a data-cite="!HR-TIME-2#dom-performance-now"><code>now()</code></a> method.</li>
+          <li>Set <var>entry</var>'s <code>duration</code> attribute to <code>0</code>.</li>
+          <li><a data-cite="!PERFORMANCE-TIMELINE-2#dfn-queue-a-performanceentry">Queue</a> <var>entry</var>.</li>
+          <li id="stored_mark">Add <var>entry</var> to the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a>.</li>
+          <li>Return <strong>undefined</strong>.</li>
+        </ol>
+      </section>
+      <section>
+        <h2><dfn>clearMarks()</dfn> method</h2>
+        <p>Removes the stored timestamp with the associated name. It MUST run these steps:</p>
+        <ol>
+          <li>If <var>markName</var> is omitted, remove all <a>PerformanceMark</a> objects from the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a>.</li>
+          <li>Otherwise, remove all <a>PerformanceMark</a> objects listed in the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a> whose <var>name</var> matches<var>markName</var>.</li>
+          <li>Return <strong>undefined</strong>.</li>
+        </ol>
+      </section>
+      <section>
+        <h2><dfn>measure()</dfn> method</h2>
+        <p>Stores the <code>DOMHighResTimeStamp</code> duration between two marks along with the associated name (a "measure"). It MUST run these steps:</p>
+        <ol>
+          <li>Run the <a>ensure name is valid</a> algorithm with <var>name</var> set to <var>measureName</var>.</li>
+          <li>Let <var>end time</var> be <code>0</code>.</li>
+          <li>If <var>endMark</var> is omitted, let <var>end time</var> be the value that would be returned by the <code>Performance</code> object's <a data-cite="!HR-TIME-2#dom-performance-now"><code>now()</code></a> method. Otherwise:
+            <ol>
+              <li>If <var>endMark</var> has the same name as a <a data-cite="!WEBIDL#dfn-read-only">read only attribute</a> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface, let <var>end time</var> be the value returned by running the <a>convert a name to a timestamp</a> algorithm with <var>name</var> set to the value of <var>endMark</var>.</li>

```

Comment:
Current logic will throw a SyntaxError inside a Worker if you use NavigationTiming attribute name. Sanity check: is this intentional?

Reply:
Yes, this is what we decided in https://github.com/w3c/user-timing/issues/22.